src/borg/item.pyx

```diff
@@ -251,6 +258,17 @@ class Item(PropDict):
     def from_optr(self, optr):
         return _optr_to_object(optr)
 
+    def is_link(self):
+        return self._is_type('l')
+
+    def is_dir(self):
+        return self._is_type('d')
+
+    def _is_type(self, typestr):
+        try:
+            return stat.filemode(self.mode)[0] == typestr
+        except AttributeError:
+            return False
```

Comment:
this is like the code previously did it, we can change this later to directly check self.mode, without the stat.filemode call.